Assets/Scripts/Army/Army.cs

```diff
@@ -433,25 +445,86 @@ private void CmdAttack(Entity entity)
             Debug.LogError("Entity has no health component");
             return;
         }
-        // Debug.Log("Attacking");
         attackTarget = entity;
         attackTarget.GetComponent<EntityHealth>().OnDie.AddListener(HandleAttackTargetOnDie);
         SetState(ArmyState.Attacking);
     }
+
+    /// <summary>
+    /// Called in the update loop when the army is attacking
+    /// </summary>
+    [Server]
+    private void Attack()
+    {
+        if (attackTarget == null) // Target died
+        {
+            // Debug.Log("Target died");
+            state = ArmyState.Idle;
+        }
+        else
+        {
+            if (Vector3.Distance(transform.position, attackTarget.transform.position) <= attackRange)
+            {
+                entityMovement.Stop();
+                attackTarget.EntityHealth.TakeDamage(attackDamage * Time.deltaTime);
+            }
+            else
+            {
+                entityMovement.Move(attackTarget.transform.position);
+            }
+        }
+    }
+
+    [Command]
+    public void CmdConvert(Entity entity)
+    {
+        if (entity == null) { return; }
+        if (entity.GetComponent<Army>() == null)
+        {
+            return;
+        }
+        convertTarget = entity;
+        convertArmy = entity.GetComponent<Army>();
+        SetState(ArmyState.Converting);
+    }
+
+    [Server]
+    private void Convert()
+    {
+        if (convertTarget == null)
+        {
+            state = ArmyState.Idle;
+            return;
+        }
+        else
+        {
+            if (Vector3.Distance(transform.position, convertTarget.transform.position) <= attackRange)
+            {
+                entityMovement.Stop();
+                if (convertArmy == null)
+                {
+                    Debug.LogError("Convert army is null");
+                }
+                else if (convertArmy.ArmyConversion == null)
+                {
+                    Debug.LogError("Convert army has no conversion component");
+                }
+                convertArmy.ArmyConversion.Convert(this);
```

Comment:
Add a method that determines which army (or neither, but this shouldn't happen according to current design) will be converted: reverse army conversion roles iff this army has fewer units than other army